src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -384,6 +395,124 @@ type Currency {
   digits: Int!
 }
 
+"Stoptime represents the time when a specific trip on a specific date arrives to and/or departs from a specific stop."
+type DatedStopTime {
+  "Scheduled arrival time to the stop and a realtime estimate, if one exists."
+  arrival: ArrivalDepartureTime
+  "Scheduled departure time from the stop and a realtime estimate, if one exists."
+  departure: ArrivalDepartureTime
+  """
+  Whether the vehicle can be disembarked at this stop. This field can also be
+  used to indicate if disembarkation is possible only with special arrangements.
+  """
+  dropoffType: PickupDropoffType
+  """
+  Vehicle headsign of the trip on this stop. Trip headsigns can change during
+  the trip (e.g. on routes which run on loops), so this value should be used
+  instead of `tripHeadsign` to display the headsign relevant to the user.
+  """
+  headsign(
+    """
+    If translated headsign is found from gtfs translation.txt and wanted language is not same as
+    feed's language then returns wanted translation. Otherwise uses name from trip_headsign.txt.
+    """
+    language: String
+  ): String
+  """
+  Whether the vehicle can be boarded at this stop. This field can also be used
+  to indicate if boarding is possible only with special arrangements.
+  """
+  pickupType: PickupDropoffType
+  "Whether stop has been updated through a realtime update and if so, how."
+  realtimeState: StopRealTimeState!
+  "The stop where this arrival/departure happens"
+  stop: Stop
+  """
+  The sequence of the stop in the pattern. This is not required to start from 0 or be consecutive - any
+  increasing integer sequence along the stops is valid.
+    
+  The purpose of this field is to identify the stop within the pattern so it can be cross-referenced
+  between it and the itinerary. It is safe to cross-reference when done quickly, i.e. within seconds.
+  However, it should be noted that real-time updates can change the values, so don't store it for
+  longer amounts of time.
+    
+  Depending on the source data, this might not be the GTFS `stop_sequence` but another value, perhaps
+  even generated.
+  """
+  stopPosition: Int
+  "true, if this stop is used as a time equalization stop. false otherwise."
+  timepoint: Boolean
+}
+
+"Trip on a specific date"
+type DatedTrip implements Node {
```

Comment:
Should we have some field(s) on this which tell if the trip has been canceled with realtime or with a scheduled cancellation?

Reply:
The ` DatedTrip` is very similar to a `DatedServiceJourney` in Netex - I would prefer if we made this the same. I think maybe you should consider comparing this with the `DatedServiceJourney`, see the doc [here](https://api.staging.entur.io/graphql-explorer/journey-planner-v3) (You need to navigate in the doc to the right place, should not be har to find).

**Important parts**
 - `stops -> quays(first, last)`
 - `stoptimes` -> `estimatedCalls`
 
**Aggregated field**
I looked at some of the materialized/computed fields in this class and I am not sure why you added them. It is good to add methods/fields if they "do" something that is defined by some business rule. This is a GraphQL best practice - but I can´t see the business rules behind the following fields - so at least doc explaining that would be good. Saving the client from navigation to the right place is NOT a business rule and should only be added if the field is "always" used, and the relation is not. This apply to:

 - start/end -> Should be alias to get the first and last call - not time. Unsure if they are needed.
 - route -> trip.route ?
 - tripHeadSign -> navigate via pattern ?
 - tripShortName -> trip.shortName ?

Reply:
This is now renamed to be FixedTripOnServiceDate (and there is an union called TripOnServiceDate). Added trip to this type and removed some fields which exist in trip.